tests/reftests/pin.test

```diff
@@ -1057,24 +1070,100 @@ echo "another-inexistant" "inexistant"
 -> installed bar.dev
 Done.
 ### :XI:c: Test opam pin remove <pkg>.<version>
-### opam pin remove --no-action bar.dev
-Ok, bar is no longer pinned to file://${BASEDIR}/bar (version dev)
+### opam switch create removal --empty
+### opam pin ./nip --no-action
+nip is now pinned to file://${BASEDIR}/nip (version ved)
+### opam pin remove --no-action nip.dev
```

Comment:
This command is duplicated with the nip.wrong-version test below
```suggestion
```

Reply:
In fact, it is 
```
### opam pin remove --no-action nip.ved
```
The logic
* test with a good version of pinned packages
* test with a wrong version of pinned packages
* test with a not pinned package

Reply:
it's not. It was:
* remove good version of pkg1
* remove wrong version of pkg2

Thus we should simplify into:
* remove wrong version of pkg1
* remove good version of pkg1

Reply:
In the original test i agree it was : remove good version then remove wrong version. I've updated the test to add also removal of unpinned one.

"The logic" i was talking about is on the wanted final test (my modif), not the original.

Reply:
i don't think the extra test makes sense and i don't think we should add new tests here. The commit changing this part is here just to add separation not adding extra tests. It's already complicated to review changes in the reftests as it is.

Reply:
In the final form (not yet pushed, i'm still debugging cygwin), i've separated the addition of unpinned package test in first commit like that the second commit only contains the switch change.